### DISCUSSION TITLE
Silently truncation of arrays when mapping to `std::array` of different size

### DISCUSSION THREAD

**Discussion by mlund** [nlohmann/json#3010] [Q&A]
The following code silently truncates a longer json array to fit the `std::array` given as default to `value()`:

``` cpp
#include <nlohmann/json.hpp>
#include <array>
#include <iostream>

int main() {
    nlohmann::json j = {{"range", std::array<int, 4>({1,2,3,4})}}; // json array too long...
    auto [min, max] = j.value("range", std::array<int, 2>({0,0})); // ...since size=2 expected
    std::cout << min << " " << max << "\n"; // -- > 1 2
}
```

Given that `std::array` is of fixed length, I would have thought that a runtime error would be generated if the given array doesn't fit?

**Comment by nlohmann** [comment]
There is a PR #2558 to fix this issue. There will be a preprocessor symbol to enable this behavior though, because otherwise we would break existing implementations.
